api/codelists.go

```diff
@@ -14,13 +16,32 @@ func (c *CodeListAPI) getCodeLists(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	filterBy := r.URL.Query().Get("type")
 
+	// get limit from query parameters, or default value
+	limit, err := utils.GetPositiveIntQueryParameter(r.URL.Query(), "limit", c.defaultLimit)
+	if err != nil {
+		log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR)
+		handleError(ctx, "failed to obtain limit", log.Data{"limit": limit}, err, w)
+		return
+	}
+
+	// get offset from query parameters, or default value
+	offset, err := utils.GetPositiveIntQueryParameter(r.URL.Query(), "offset", c.defaultOffset)
```

Comment:
Try to avoid having a generic `utils` package, as it can become a dumping ground for lots of unrelated functions. In this case I would consider having the query parameter function defined in `api.go`, with a longer term view of maybe moving it to the dp-net library.